packages/cozy-stack-client/src/memoize.spec.js

```diff
@@ -19,4 +20,23 @@ describe('memoize', () => {
     expect(mcounter()).toBe(1)
     expect(mcounter()).toBe(1)
   })
+
+  it('should not memoize the result if an error occurs', async () => {
+    now = 0
+    c = 0
+    const counter = async () => {
+      c++
+      return new Promise(function(resolve, reject) {
+        resolve(new ErrorReturned())
+      })
+    }
+    const mcounter = memoize(counter, { key: () => 'A', maxDuration: 5 })
+    mcounter()
+    expect(c).toEqual(1)
+    mcounter()
+    //Let's wait the promise resolution
+    setTimeout(() => {
+      expect(c).toEqual(2)
+    }, 10)
```

Comment:
I don't know how to do differently without this timeout since we're waiting the promise resolution before deleting the cache...

Reply:
Since the expect is in a timeout, it is not called before the test finishes.

Better to await a sleep for this.

```

  const sleep = duration =>
    new Promise(resolve => setTimeout(resolve, duration))

    await sleep(12)
```

The second expect does not pass currently.